sync.js

```diff
@@ -62,6 +62,8 @@ GlobSync.prototype._finish = function () {
         } catch (er) {
           if (er.syscall === 'stat')
             set[self._makeAbs(p)] = true
+          else if (er.code === 'ELOOP')
+            if (real) set[real] = true
           else
```

Comment:
Also, how can `real` be set if an error occurred?  Does `fs.realpath` ever return a value if it raises an error?  That seems really strange.